app/models/product.rb

```diff
@@ -22,10 +22,17 @@ def validate_each(record, attribute, value)
   end
```

Comment:
@avneetsng List down all the callbacks called when we do the following steps.

1. Create a Product.
  a. Product.new
  b. Product.save
  
2. Update a Product
  a. p = Product.first
  b. p.save

3. Destroy a product
   a. p = Product.first
   b. p.destroy

Reply:
Ans for 1st by @avneetsng 
p = Product.new
 1. after_initialize

p.save
 1. before_validation
 2. after_validation
 3. before_save
 4. after_save
 5. before_create
 6. around_update
 7. after_update
 8. after_save
 9. after_commit